src/testing/covered.zig

```diff
@@ -0,0 +1,124 @@
+//! We'll get you covered!
+//!
+//! This file piggy-backs on the logging infrastructure to implement explicit coverage marks:
+//!     <https://ferrous-systems.com/blog/coverage-marks/>
+//!
+//! In production code, you can mark certain log lines as "this should be covered by a test".
+//! In test code, you can then assert that a _specific_ test covers a specific log line. The two
+//! benefits are:
+//! - tests are more resilient to refactors
+//! - production code is more readable (you can immediately jump to a specific test)
+test "tutorial" {
```

Comment:
Was inspired by https://github.com/ziglang/zig/pull/18955/commits/00acf8a66dec78743d26d62e86391b8794a1c627 to just put a "tutorial" style test right after the doc comment, instead of doing inline examples.